integration-tests/bats/import-mysqldump.bats

```diff
@@ -0,0 +1,241 @@
+
+-- Dump completed on 2022-04-12 14:00:34
+SQL
+    [ "$status" -eq 0 ]
+
+    run dolt sql -q "SHOW DATABASES"
+    [[ "$output" =~ "test" ]] || false
+}
+
+@test "import-mysqldump: import a simple database" {
+    run dolt sql <<SQL
+DROP TABLE IF EXISTS test;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE test (
+  pk int(11) NOT NULL,
+  c1 varchar(32) DEFAULT NULL,
+  PRIMARY KEY (pk)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
+/*!40101 SET character_set_client = @saved_cs_client */;
+
+LOCK TABLES test WRITE;
+/*!40000 ALTER TABLE test DISABLE KEYS */;
+INSERT INTO test VALUES (0,'foo'),(1,'bar');
+/*!40000 ALTER TABLE test ENABLE KEYS */;
+UNLOCK TABLES;
+SQL
+    [ "$status" -eq 0 ]
+
+    run dolt sql -q "SELECT * FROM test" -r csv
+    [ "$output" = "pk,c1
+0,foo
+1,bar" ]
+}
+
+@test "import-mysqldump: import a database with view" {
+    run dolt sql <<SQL
+DROP TABLE IF EXISTS mytable;
+CREATE TABLE mytable (
+  id bigint NOT NULL,
+  col2 bigint DEFAULT '999',
+  col3 datetime DEFAULT CURRENT_TIMESTAMP,
+  PRIMARY KEY (id)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
+
+LOCK TABLES mytable WRITE;
+/*!40000 ALTER TABLE mytable DISABLE KEYS */;
+/*!40000 ALTER TABLE mytable ENABLE KEYS */;
+UNLOCK TABLES;
+
+--
+-- Temporary view structure for view myview
+--
+
+DROP TABLE IF EXISTS myview;
+/*!50001 DROP VIEW IF EXISTS myview*/;
+/*!50001 CREATE VIEW myview AS SELECT
+ 1 AS id,
+ 1 AS col2,
+ 1 AS col3*/;
+
+--
+-- Final view structure for view myview
+--
+
+/*!50001 DROP VIEW IF EXISTS myview*/;
+/*!50001 CREATE ALGORITHM=UNDEFINED */
+/*!50013 DEFINER=root@localhost SQL SECURITY DEFINER */
+/*!50001 VIEW myview AS select mytable.id AS id,mytable.col2 AS col2,mytable.col3 AS col3 from mytable */;
+SQL
+    [ "$status" -eq 0 ]
+
+    dolt sql -q "INSERT INTO mytable (id, col3) VALUES (1, TIMESTAMP('2003-12-31'));"
+    run dolt sql -q "SELECT * FROM myview;"
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "| 1  | 999  | 2003-12-31 00:00:00 +0000 UTC |" ]] || false
+}
+
+@test "import-mysqldump: import a database with trigger" {
+    run dolt sql <<SQL
+DROP TABLE IF EXISTS mytable;
+CREATE TABLE mytable (
+  pk bigint NOT NULL,
+  v1 bigint DEFAULT NULL,
+  PRIMARY KEY (pk)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
+
+LOCK TABLES mytable WRITE;
+/*!40000 ALTER TABLE mytable DISABLE KEYS */;
+INSERT INTO mytable VALUES (0,2),(1,3),(2,44);
+/*!40000 ALTER TABLE mytable ENABLE KEYS */;
+UNLOCK TABLES;
+
+/*!50003 SET @saved_sql_mode       = @@sql_mode */ ;
+/*!50003 SET sql_mode              = 'ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION' */ ;
+DELIMITER ;;
+/*!50003 CREATE*/ /*!50017 DEFINER=root@localhost*/ /*!50003 TRIGGER tt BEFORE INSERT ON mytable FOR EACH ROW SET NEW.v1 = NEW.v1 * 11 */;;
```

Comment:
Can you add backticks to `root` and `localhost` as part of the definer. cc: https://github.com/dolthub/dolt/issues/3091